core/payment-driver/src/gnt/sender.rs

```diff
@@ -689,3 +781,46 @@ impl TransactionSender {
         ctx.spawn(job);
     }
 }
+
+pub struct AccountLocked {
+    pub identity: NodeId,
+}
+
+impl Message for AccountLocked {
+    type Result = Result<(), PaymentDriverError>;
+}
+
+impl Handler<AccountLocked> for TransactionSender {
+    type Result = ActorResponse<Self, (), PaymentDriverError>;
+
+    fn handle(&mut self, msg: AccountLocked, _ctx: &mut Self::Context) -> Self::Result {
+        log::info!("Account: {:?} is locked", msg.identity.to_string());
+        self.active_accounts
+            .borrow_mut()
+            .remove_account(msg.identity);
+        ActorResponse::reply(Ok(()))
+    }
+}
+
+pub struct AccountUnlocked {
+    pub identity: NodeId,
+}
+
+impl Message for AccountUnlocked {
+    type Result = Result<(), PaymentDriverError>;
+}
+
+impl Handler<AccountUnlocked> for TransactionSender {
+    type Result = ActorResponse<Self, (), PaymentDriverError>;
+
+    fn handle(&mut self, msg: AccountUnlocked, _ctx: &mut Self::Context) -> Self::Result {
+        log::info!("Account: {:?} is unlocked", msg.identity.to_string());
```

Comment:
maybe its better to call `add_account(`before logging?  in case something fails the logs are inacurate

Reply:
:+1: